include/deal.II/multigrid/mg_transfer_global_coarsening.templates.h

```diff
@@ -68,7 +68,9 @@ namespace
     bool
     run(Fu &fu)
     {
-      if ((degree_fine == (2 * deg + 1)) && (degree_coarse == deg))
+      if ((degree_fine == (2 * deg)) && (degree_coarse == deg))
+        fu.template run<2 * deg, deg>(); // h-MG (FE_Q)
```

Comment:
Isn't this the case
```suggestion
        fu.template run<2 * deg, deg>(); // h-MG (FE_DGQ)
```
and `FE_Q` is the case in line 74?

Reply:
Why should it? DG has an additional point in the middle. Also the case in line 74 is not just DG.

Reply:
That brings us back to the question: What is `deg`? With what you're saying, it is not directly the number of points, because a continuous element cannot have `2*deg` and `deg` points in an h-MG setting, i.e., due to the odd number of points of FE_Q on the fine side. So my interpretation is that the values you give here are not the number of points, but not really a degree either, but really the number of points - 1 on both ends, implying an "equivalent" higher degree on the refined side. Again, this reasoning of a degree breaks down for DG, and it seems that this merely is a parameter collecting some information that gets interpreted in the right way inside the function. (This is also what you explicitly hardcode below, so we are really in the need of adjusting variable names soon as you suggest in the other reply below.)